flow/Tracing.actor.cpp

```diff
@@ -242,6 +283,67 @@ struct UDPTracer : public ITracer {
 		}
 	}
 
+	// Writes the given vector of linked SpanContext's to the request. If the vector is
+	// empty, the request is not modified.
+	inline void serialize_vector(const SmallVectorRef<SpanContext>& vec, TraceRequest& request) {
+		int size = vec.size();
+		if (size <= 15) {
+			request.write_byte(static_cast<uint8_t>(size) | 0b10010000);
+		} else if (size <= 65535) {
+			request.write_byte(0xdc);
+			request.write_byte(reinterpret_cast<const uint8_t*>(&size)[1]);
+			request.write_byte(reinterpret_cast<const uint8_t*>(&size)[0]);
+		} else {
+			// TODO: Add support for longer vectors if necessary.
+			ASSERT(false);
+		}
+
+		for (const auto& link : vec) {
+			serialize_value(link.traceID.first(), request, 0xcf); // trace id
+			serialize_value(link.traceID.second(), request, 0xcf); // trace id
+			serialize_value(link.spanID, request, 0xcf); // spanid
+		}
+	}
+
+	// Writes the given vector of linked SpanContext's to the request. If the vector is
+	// empty, the request is not modified.
+	inline void serialize_vector(const SmallVectorRef<OTELEvent>& vec, TraceRequest& request) {
+		int size = vec.size();
+		if (size <= 15) {
+			request.write_byte(static_cast<uint8_t>(size) | 0b10010000);
+		} else if (size <= 65535) {
+			request.write_byte(0xdc);
+			request.write_byte(reinterpret_cast<const uint8_t*>(&size)[1]);
+			request.write_byte(reinterpret_cast<const uint8_t*>(&size)[0]);
+		} else {
+			// TODO: Add support for longer vectors if necessary.
+			ASSERT(false);
+		}
+
+		for (const auto& event : vec) {
+			serialize_string(event.name.toString(), request); // event name
+			serialize_value(event.time, request, 0xcb); // event time
+			serialize_vector(event.attributes, request);
+		}
+	}
+
+	inline void serialize_vector(const SmallVectorRef<KeyValueRef>& vals, TraceRequest& request) {
+		int size = vals.size();
+		if (size <= 15) {
+			// N.B. We're actually writing this out as a fixmap here in messagepack format!
+			// fixmap	1000xxxx	0x80 - 0x8f
+			request.write_byte(static_cast<uint8_t>(size) | 0b10000000);
+		} else {
+			// TODO: Add support for longer maps if necessary.
+			ASSERT(false);
```

Comment:
Alternatively you can use `ASSERT_WE_THINK(false)`. This will only cause a crash in simulation but not in prod. Additionally I would still log something (just to make debugging easier should we ever run into this issue).